src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/RuntimeFunctionsTableNode.cs

```diff
@@ -103,12 +103,17 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             return runtimeFunctionsBuilder.ToObjectData();
         }
 
-        public int TableSize
+        /// <summary>
+        /// Returns the runtime functions table size and excludes the 4 byte sentinel entry at the end (used by
+        /// the runtime in NativeUnwindInfoLookupTable::LookupUnwindInfoForMethod) so that it's not treated as
+        /// part of the table itself.
+        /// </summary>
+        public int TableSizeExcludingSentinel
         {
             get
             {
                 Debug.Assert(_tableSize >= 0);
-                return _tableSize;
+                return _tableSize - 4;
```

Comment:
Nit - it's kind of awkward that we need to duplicate this logic between this place and the section table emitter. Perhaps we might be able to shove some virtual method like <code>EndSentinelSize</code> or <code>SizeDelta</code> into HeaderTableNode and use it for these adjustments. Having said that, I know you're running out of time so please feel free to ignore it, it's no big deal and we can clean this up later if we decide it's worth it.

Reply:
Good idea - I'll add a SentinelSize property so it's more descriptive.

Reply:
Well, actually it's more awkward than I thought. Somewhere along the way, we stopped treating header table component nodes as `HeaderTableNode`s and in fact I further generalized it to `DependencyNodeCode<NodeFactory>` in this commit. For now I'll replace the numbers with a const in `RuntimeFunctionsTableNode`. `HeaderTableNode` currently just provides a base to factor default values for `ObjectNode` required overrides. We might want to just refactor it away altogether and use an interface on the header table component nodes to get the small amount of functionality needed.

Reply:
Sounds good, thank you!